shared/actions/deeplinks.tsx

```diff
@@ -0,0 +1,114 @@
+import * as ChatGen from './chat2-gen'
+import * as Constants from '../constants/config'
+import * as Container from '../util/container'
+import * as DeeplinksGen from './deeplinks-gen'
+import * as ProfileGen from './profile-gen'
+import * as RPCTypes from '../constants/types/rpc-gen'
+import * as RouteTreeGen from './route-tree-gen'
+import * as Saga from '../util/saga'
+import * as Tabs from '../constants/tabs'
+import * as WalletsGen from './wallets-gen'
+import URL from 'url-parse'
+
+const handleKeybaseLink = (_: Container.TypedState, action: DeeplinksGen.HandleKeybaseLinkPayload) => {
+  const error =
+    "We couldn't read this link. The link might be bad, or your Keybase app might be out of date and need to be updated."
+  const parts = action.payload.link.split('/')
+  // List guaranteed to contain at least one elem.
+  switch (parts[0]) {
+    case 'profile':
+      if (parts.length !== 3 && parts.length !== 4) {
+        return [
+          DeeplinksGen.createSetKeybaseLinkError({error}),
+          RouteTreeGen.createNavigateAppend({
+            path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+          }),
+        ]
+      }
+      if (parts[1] === 'new-proof') {
+        return [
+          parts.length === 4 && parts[3] ? ProfileGen.createUpdateUsername({username: parts[3]}) : null,
+          ProfileGen.createAddProof({platform: parts[2], reason: 'appLink'}),
+        ]
+      }
+      break
+    // Fall-through
+    case 'private':
+    case 'public':
+    case 'team':
+      return [
+        RouteTreeGen.createSwitchTab({tab: Tabs.fsTab}),
+        RouteTreeGen.createNavigateAppend({
+          path: [{props: {path: '/keybase/' + parts.join('/')}, selected: 'main'}],
```

Comment:
Could just do `/keybase/${action.payload.link}` here instead of joining?

Reply:
Thanks, done.